### ISSUE TITLE
In Blazor server-side project if .razor component is in /Shared folder then it is not found by F12 and refactoring

### ISSUE THREAD

**Issue by KrystofZacek** [dotnet/aspnetcore#21780]
In Blazor server-side project if .razor component is in /Shared folder then neither the component nor its members (methods, subclasses) are found by F12 and also refactoring works weirdly on them - see below:

Say I have Counter.razor component in /Shared.
The component has a public method SetCount(int count).
I can happily use the component from Pages/Index.razor.

@page "/"
<Counter @ref="counter" />
<button @onclick="OnClick">Set counter to 33</button>

@code {
    Counter counter;
    void OnClick()
    {
        counter.SetCount(33);
    }
}

But from Index.razor I cannot F12 to neither Counter class nor SetCount receiving message "Cannot navigate to symbol under carret".
Moreover - if I refactor the SetCount method from the calling Index.razor then the method name changes in Index.razor but NOT in Counter.razor leading to compilation failure.
If the Counter.razor is placed under /Pages then all works fine.

VS 2019 16.5.5
.Net Core SDK 3.1.202

**Comment by mzhukovs** [comment]
Yes I just tried Blazor today and was expecting that to work, it would be very convenient. Hope the team can include it  @mkArtakMSFT

**Comment by Jim100101** [comment]
There is a free Visual Studio Extension on Visual Studio Marketplace to accomplish Razor Code Method Navigation when editing a razor file in Visual Studio.
https://marketplace.visualstudio.com/items?itemName=VarxSystems.RazorCodeNavigator
